pkg/name/ref.go

```diff
@@ -48,9 +48,27 @@ func ParseReference(s string, opts ...Option) (Reference, error) {
 
 }
 
-// MustParseReference behaves like ParseReference, but panics instead of returning an error.
-func MustParseReference(s string, opts ...Option) Reference {
-	ref, err := ParseReference(s, opts...)
+type stringConst string
+
+// MustParseReference behaves like ParseReference, but panics instead of
+// returning an error. It's intended for use in tests, or when a value is
+// expected to be valid at code authoring time.
+//
+// To discourage its use in scenarios where the value is not known at code
+// authoring time, it must be passed a string constant:
+//
+//    MustParseReference("valid/string")
```

Comment:
Just because `prefix+testCase` is common enough, how about:

```
const validStr = "valid/string"
MustParseReference(validStr)
MustParseReference("a/second/"+validStr)
MustParseReference("yet/another/valid/string")
```